services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/PushMonitorUtils.java

```diff
@@ -59,15 +60,17 @@ public static ExecutionStatusWithDetails getDaVinciPushStatusAndDetails(
       totalReplicaCount += instances.size();
       for (Map.Entry<CharSequence, Integer> entry: instances.entrySet()) {
         ExecutionStatus status = ExecutionStatus.fromInt(entry.getValue());
+        if (status == completeStatus) {
+          completedReplicaCount++;
+          continue;
+        }
+        // For live replica change, we will skip completed replicas.
         boolean isInstanceAlive = reader.isInstanceAlive(storeName, entry.getKey().toString());
         if (!isInstanceAlive) {
           continue;
         }
         // We only compute status based on live instances.
         liveReplicaCount++;
-        if (status == completeStatus) {
-          continue;
-        }
         if (status == middleStatus) {
```

Comment:
should we move this check before continue?

Reply:
Hmm I think that's what I am doing in this PR

Reply:
we stop heartbeat in `endOfIncrementalPushReceived` as well, so it will show up as not live and we will then continue earlier before coming here.

Reply:
```
ExecutionStatus completeStatus = incrementalPushVersion.isPresent()
        ? ExecutionStatus.END_OF_INCREMENTAL_PUSH_RECEIVED
        : ExecutionStatus.COMPLETED;
```
`completeStatus` is different when incremental push is enabled.